File: examples/playwright/node/quickstart.mjs
/**
 * BotCloud Quick Start - Playwright (Node.js)
 *
 * This shows how to connect to BotCloud using Playwright's CDP connection.
 *
 * Prerequisites:
 *   npm install playwright
 *
 * Usage:
 *   1. Modify the CONFIG section below with your token and proxy
 *   2. Run: node examples/quickstart/playwright-node.mjs
 */

// ============ Configuration (Modify these) ============
const CONFIG = {
  token: "your-token-here",
  proxy: "username:password@proxy.example.com:4600",
  deviceType: "mac" // or "win", "android"
};
// ======================================================

import { chromium } from 'playwright';

function buildEndpoint() {
  if (!CONFIG.token || !CONFIG.proxy) {
    throw new Error('Please configure TOKEN and PROXY in the CONFIG section above');
  }

  const params = new URLSearchParams({
    token: CONFIG.token,
    '--proxy-server': CONFIG.proxy,
    device_type: CONFIG.deviceType,
  });
  return `wss://cloud.bots.win?${params.toString()}`;
}

async function main() {
  console.log('Connecting to BotCloud...');
  const browser = await chromium.connectOverCDP(buildEndpoint());

  console.log('Connected! Getting context...');
  const context = browser.contexts()[0] || await browser.newContext();
  const page = await context.newPage();

  console.log('Navigating to example.com...');
  await page.goto('https://example.com');

  console.log('Taking screenshot...');
  await page.screenshot({ path: 'screenshot.png' });

  console.log('Closing browser...');
  await browser.close();

  console.log('✅ Done! Check screenshot.png');
}

main().catch((error) => {
  console.error('❌ Error:', error.message);
  process.exit(1);
});

File: examples/playwright/python/quickstart.py
"""BotCloud Quick Start - Playwright (Python)

This shows how to connect to BotCloud using Playwright's async API.

Prerequisites:
    pip install playwright
    playwright install chromium

Usage:
    1. Modify the CONFIG section below with your token and proxy
    2. Run: python examples/quickstart/playwright-python.py
"""

# ============ Configuration (Modify these) ============
CONFIG = {
    "token": "your-token-here",
    "proxy": "username:password@proxy.example.com:4600",
    "device_type": "mac"  # or "win", "android"
}
# ======================================================

import asyncio
import urllib.parse
from playwright.async_api import async_playwright


def build_endpoint() -> str:
    """Build BotCloud WebSocket endpoint URL."""
    if not CONFIG["token"] or not CONFIG["proxy"]:
        raise RuntimeError("Please configure token and proxy in CONFIG section above")

    params = urllib.parse.urlencode({
        "token": CONFIG["token"],
        "--proxy-server": CONFIG["proxy"],
        "device_type": CONFIG["device_type"],
    })
    return f"wss://cloud.bots.win?{params}"


async def main() -> None:
    """Main execution function."""
    print("Connecting to BotCloud...")
    async with async_playwright() as playwright:
        browser = await playwright.chromium.connect_over_cdp(build_endpoint())

        print("Connected! Getting context...")
        context = browser.contexts[0] if browser.contexts else await browser.new_context()
        page = await context.new_page()

        print("Navigating to example.com...")
        await page.goto("https://example.com")

        print("Taking screenshot...")
        await page.screenshot(path="screenshot.png")

        print("Closing browser...")
        await browser.close()

        print("✅ Done! Check screenshot.png")


if __name__ == "__main__":
    asyncio.run(main())
